crates/rolldown/tests/rolldown/topics/hmr/runtime_correctness/artifacts.snap

```diff
@@ -0,0 +1,251 @@
+		init_rolldown_runtime_4();
+		const hot_rolldown_hmr = __rolldown_runtime__.createModuleHotContext("rolldown:hmr");
+		var import_rolldown_runtime_0 = __rolldown_runtime__.loadExports("rolldown:runtime");
+		class Module {
+			/**
+			* @type {any}
+			*/
+			exports = null;
+			/**
+			* @type {string}
+			*/
+			id;
+			/**
+			* @param {string} id
+			*/
+			constructor(id) {
+				this.id = id;
+			}
+		}
+		class DevRuntime {
+			/**
+			* @type {Record<string, Module>}
+			*/
+			modules = {};
+			/**
+			* @param {string} _moduleId
+			*/
+			createModuleHotContext(_moduleId) {
+				throw new Error("createModuleHotContext should be implemented");
+			}
+			/**
+			* @param {string[]} _boundaries
+			*/
+			applyUpdates(_boundaries) {
+				throw new Error("applyUpdates should be implemented");
+			}
+			/**
+			* @param {string} id
+			* @param {{ exports: any }} meta
+			*/
+			registerModule(id, meta) {
+				const module = new Module(id);
+				module.exports = meta.exports;
+				this.modules[id] = module;
+			}
+			/**
+			* @param {string} id
+			*/
+			loadExports(id) {
+				const module = this.modules[id];
+				if (module) {
+					return module.exports;
+				} else {
+					console.warn(`Module ${id} not found`);
+					return {};
+				}
+			}
+			/**
+			* __esmMin
+			*
+			* @type {<T>(fn: any, res: T) => () => T}
+			* @internal
+			*/
+			createEsmInitializer = (fn, res) => () => (fn && (res = fn(fn = 0)), res);
+			/**
+			* __commonJSMin
+			*
+			* @type {<T extends { exports: any }>(cb: any, mod: { exports: any }) => () => T}
+			* @internal
+			*/
+			createCjsInitializer = (cb, mod) => () => (mod || cb((mod = { exports: {} }).exports, mod), mod.exports);
+			/** @internal */
+			__toESM = import_rolldown_runtime_0.__toESM;
+			/** @internal */
+			__toCommonJS = import_rolldown_runtime_0.__toCommonJS;
+			/** @internal */
+			__export = import_rolldown_runtime_0.__export;
+			/** @internal */
+			__toDynamicImportESM = import_rolldown_runtime_0.__toDynamicImportESM;
+			/** @internal */
+			__reExport = import_rolldown_runtime_0.__reExport;
+		}
+		class TestDevRuntime extends DevRuntime {
+			/**
+			* @override
+			* @param {string} _moduleId
+			*/
+			createModuleHotContext(_moduleId) {
+				return { accept() {} };
+			}
+			/**
+			* @override
+			* @param {string[]} _boundaries
+			*/
+			applyUpdates(_boundaries) {}
+		}
+		globalThis.__rolldown_runtime__ ??= new TestDevRuntime();
+		/** @type {string[]} */
+		const testPatches = globalThis.__testPatches;
+		if (testPatches) {
+			setTimeout(async () => {
+				for (const patchChunk of testPatches) {
+					await import(patchChunk);
+				}
+			}, 0);
+		}
+	} finally {}
+}));
+
+//#endregion
+//#region rolldown:runtime
+var init_rolldown_runtime_4 = __rolldown_runtime__.createEsmInitializer((function() {
+	try {
+		var __rolldown_exports__ = {};
+		__rolldown_runtime__.__export(__rolldown_exports__, {});
```

Comment:
Here as well. Also seems to be included twice, one for CJS and one for ESM.

Reply:
Found the cause.